### PR TITLE
Restore deploy-storybook job

### PR DESCRIPTION
**Pull request by jerivas**
## Description
Restore the job that deploys to Storybook that was lost when migrating from CircleCI

From https://github.com/oddbird/Metecho/pull/2149#discussion_r865138094

### CODE REVIEW

**Comment by jerivas** [comment]
[Deploy Storybook docs via GH action on pushes to main branch.](https://trello.com/c/dwNddOEN/361-deploy-storybook-docs-via-gh-action-on-pushes-to-main-branch)

File: .github/workflows/deploy-storybook.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - uses: actions/checkout@v3
+      - name: Determine node version
+        run:
+          echo "NODE_VERSION=$(grep -oP 'node\K\d+' Dockerfile)" >> $GITHUB_ENV
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '${{ env.NODE_VERSION }}'
+          cache: yarn
+      - run: yarn install --frozen--lockfile
+      - run: yarn deploy-storybook --ci
```
Comment (jerivas) [inline .github/workflows/deploy-storybook.yml:24]: @jgerigmeyer The original job wrapped this in an if:

```bash
if [ -n "${DEPLOY_STORYBOOK}" ]; then
   yarn deploy-storybook --ci
fi
```
Is that necessary? I couldn't find where that env var was set

File: .github/workflows/deploy-storybook.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Determine node version
+        run:
+          echo "NODE_VERSION=$(grep -oP 'node\K\d+' Dockerfile)" >> $GITHUB_ENV
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '${{ env.NODE_VERSION }}'
+          cache: yarn
```
Comment (jgerigmeyer) [inline .github/workflows/deploy-storybook.yml:22]: I'm guessing this is just quicker than using our actual container via `setup-docker-compose`?

File: .github/workflows/deploy-storybook.yml (hunk 1 above)
Comment (jgerigmeyer) [reply .github/workflows/deploy-storybook.yml:24]: Yeah, that was set directly on our CircleCI, with the goal of this step being skipped on SFDO's CircleCI runs without officially failing the CI check. Let's hold off on that for now, until we have a clearer sense of the eventual hand-off path here.

**Review by jgerigmeyer — APPROVED** [review]
Looks good! I'm curious whether this will work without adding a new deploy key or something. On CircleCI we had to add an SSH key for this...

https://github.com/storybookjs/storybook-deployer#deploying-storybook-as-part-of-a-ci-service

**Comment by jgerigmeyer** [comment]
Hrm... `Canceling since a higher priority waiting request for 'refs/heads/main' exists`

https://github.com/oddbird/Metecho/actions/runs/2277924304

**Comment by jgerigmeyer** [comment]
Looks like this is the issue (possibly in addition to needing a token or PAT?): https://github.com/storybookjs/storybook-deployer/issues/77

https://github.com/oddbird/Metecho/runs/6313167269?check_suite_focus=true

File: .github/workflows/deploy-storybook.yml (hunk 2 above)
Comment (jerivas) [reply .github/workflows/deploy-storybook.yml:22]: Yep, for frontend only stuff this is faster than going through the whole docker setup process

**Pull request merged**